cocotb/share/lib/vhpi/VhpiCbHdl.cpp

```diff
@@ -438,13 +438,16 @@ int VhpiCbHdl::arm_callback()
         if (!new_hdl) {
             check_vhpi_error();
             LOG_ERROR("VHPI: Unable to register a callback handle for VHPI type %s(%d)",
-                         m_impl->reason_to_string(cb_data.reason), cb_data.reason);
+                      m_impl->reason_to_string(cb_data.reason), cb_data.reason);
             goto error;
         }
 
         cbState = (vhpiStateT)vhpi_get(vhpiStateP, new_hdl);
-        if (vhpiEnable != cbState) {
-            LOG_ERROR("VHPI: Registered callback isn't enabled! Got %d", cbState);
+        if (vhpiUndefined == cbState) {
```

Comment:
I think you need to check this before you cast the result of the call to `vhpi_get` to a `vhpiStateP`. `vhpiUndefined` is a valid result of `vhpi_get`, but it isn't in the `vhpiStateP` enum, hence the warning.

Reply:
Ah, thanks for the explanation.
The current code is not exactly nice, is there a way to improve it? Or is all that casting too much for too little gain?
What prompted this is that `vhpiUndefined` is simulator-dependent:
```
#ifndef IUS
#define vhpiUndefined -1
#else
#define vhpiUndefined 1000
#endif
```
But, oh,  I now see that I am casting `vhpiUndefined` wrong (to `unsigned`), but then, `typedef uint32_t vhpiIntT`... what a mess. I guess I'll just revert that part of the PR.